### PR TITLE
Broke out many helper methods inside of ``DefaultRavenFactory`` to al…

### PR DESCRIPTION
**Pull request by bretthoerner**
…low for easier overrides.

### CODE REVIEW

File: raven/src/main/java/com/getsentry/raven/connection/AsyncConnection.java [hunk 1]
```diff
@@ -84,7 +81,7 @@ public AsyncConnection(Connection actualConnection, ExecutorService executorServ
      * @param gracefulShutdown Indicates whether or not the shutdown operation should be managed by a ShutdownHook.
      */
     public AsyncConnection(Connection actualConnection, ExecutorService executorService, boolean gracefulShutdown) {
-        this(actualConnection, executorService, gracefulShutdown, DEFAULT_SHUTDOWN_TIMEOUT);
+        this(actualConnection, executorService, gracefulShutdown, DefaultRavenFactory.ASYNC_SHUTDOWN_TIMEOUT_DEFAULT);
```
Comment (tkaemming) [inline raven/src/main/java/com/getsentry/raven/connection/AsyncConnection.java]: It seems a little weird to grab these out of the default factory. It seems like we should either:

1.) Define the default values here, and fill them in if they either not provided to this constructor or are null in the other constructor, or,
2.) Define the default values in the default factory, and drop this shorter constructor.

(This same comment applies to the similar constructors below.)

File: raven/src/main/java/com/getsentry/raven/DefaultRavenFactory.java [hunk 2]
```diff
@@ -341,6 +315,182 @@ protected Marshaller createMarshaller(Dsn dsn) {
     }
 
     /**
+     * Returns whether or not to wrap the underlying connection in an {@link AsyncConnection}.
+     *
+     * @param dsn Dsn passed in by the user.
+     * @return whether or not to wrap the underlying connection in an {@link AsyncConnection}.
+     */
+    protected boolean getAsync(Dsn dsn) {
```
Comment (tkaemming) [inline raven/src/main/java/com/getsentry/raven/DefaultRavenFactory.java]: This method name is a little weird. Not sure what a better name would be, though. `getAsyncEnabled` or something?

File: raven/src/main/java/com/getsentry/raven/connection/AsyncConnection.java (hunk 1 above)
Comment (bretthoerner) [inline raven/src/main/java/com/getsentry/raven/connection/AsyncConnection.java]: I'm OK with 2, but wasn't sure about dropping them without deprecation. Probably not a big deal?

The reason 1 stinks is it means my `getSomething` method has no default, and so now must return `null` or some sentinel value which made me sad considering every other new getter was always expected to return a "real" value. Felt gross to me to have some that could return `null` and some that couldn't.

**Comment by bretthoerner** [comment]
Ready for review.

**Pull request merged**